src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/ShardedHttpExecutionServiceCluster.scala

```diff
@@ -0,0 +1,109 @@
+  private val memberArray:Array[ClusterMember] = (members map {case (id, dao) => ClusterMember(id, dao)})
+      .toList.sortBy(_.key.id).toArray
+
+  // find the default instance
+  private val defaultEntry:(ExecutionServiceId, ExecutionServiceDAO) =
+    members.find{case (esid, dao) => esid.id.equals(default.id)}
+      .getOrElse(throw new RawlsException(s"cannot initialize ShardedHttpExecutionServiceCluster: member with default key $default does not exist"))
+
+
+  // ====================
+  // facade methods
+  // ====================
+
+  // only used in tests??? TODO: DA update tests to use the same thing runtime does, or remove this.
+  def submitWorkflow(wdl: String, inputs: String, options: Option[String], userInfo: UserInfo): Future[ExecutionServiceStatus] =
+    getDefaultMember.submitWorkflow(wdl, inputs, options, userInfo)
+
+  // by nature, this is only called for workflows that have not yet been submitted.
+  // therefore, we want to send the workflows to the cromwell instance chosen
+  // by the routing algorithm defined in this class.
+  def submitWorkflows(workflowRecs: Seq[WorkflowRecord], wdl: String, inputs: Seq[String], options: Option[String], userInfo: UserInfo): Future[(ExecutionServiceId, Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]])] = {
+    val targetMember = targetMemberForSubmission(workflowRecs)
+    targetMember.dao.submitWorkflows(wdl, inputs, options, userInfo) map {results =>
+      (targetMember.key, results)
+    }
+  }
+
+  // following are called on a workflow that has already been submitted.
+  // therefore, we want to use the cromwell instance that has been persisted
+  // onto that workflow.
+  def status(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceStatus] =
+    getMember(wfe).status(wfe.id, userInfo)
+
+  def callLevelMetadata(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionMetadata] =
+    getMember(wfe).callLevelMetadata(wfe.id, userInfo)
+
+  def outputs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceOutputs] =
+    getMember(wfe).outputs(wfe.id, userInfo)
+
+  def logs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceLogs] =
+    getMember(wfe).logs(wfe.id, userInfo)
+
+  def abort(wfe: WorkflowExecution, userInfo: UserInfo): Future[Try[ExecutionServiceStatus]] =
+    getMember(wfe).abort(wfe.id, userInfo)
+
+
+  // ====================
+  // facade-to-cluster entry points
+  // ====================
+  // for an already-submitted workflow, get the instance to which it was submitted
+  private def getMember(wfe: WorkflowExecution):ExecutionServiceDAO = members.get(wfe.executionServiceId) match {
+    case Some(dao) => dao
+    // all legacy workflows were updated in SQL to have a key of "default". If the cluster does not contain
+    // a member with that key, use the cluster's default entry.
+    case None if ExecutionServiceCluster.DEFAULT_KEY.equals(wfe.executionServiceId) => getDefaultMember
```

Comment:
I don't think we should have a default. This kind of indirection can only lead to problems where someone innocently decides to change our default to cromwell2 which then breaks any workflows associated to the default when it was cromwell1.